lively.ide/studio/controls/popups.cp.js

```diff
@@ -745,119 +779,170 @@ const InsetShadowPopup = component(ShadowPopup, {
+  layout: new TilingLayout({
+    align: 'right',
+    axisAlign: 'center',
+    orderByIndex: true,
+    padding: rect(0, 0, 10, 0),
+    spacing: 2,
+    wrapSubmorphs: false
+  }),
+  extent: pt(230, 30),
+  fill: Color.rgba(0, 0, 0, 0),
+  submorphs: [
+    part(NumberWidgetLight, {
+      name: 'padding all',
+      min: 0,
+      number: 0,
+      extent: pt(70, 22),
+      position: pt(9.7, 6.6),
+      tooltip: 'Padding',
+      submorphs: [add({
+        type: Label,
+        name: 'interactive label',
+        fontFamily: 'Material Icons',
+        fontColor: Color.rgb(101, 135, 139),
+        padding: rect(8, 0, -1, 0),
+        textAndAttributes: ['\ue22f', {
+          fontSize: 16,
+          textStyleClasses: ['material-icons']
+        }]
+      }, 'value')]
+    }),
+    {
+      name: 'multi padding control',
+      position: pt(7.6, 33.4),
+      layout: new TilingLayout({
+        align: 'center',
+        axisAlign: 'center',
+        orderByIndex: true
+      }),
+      fill: Color.transparent,
+      extent: pt(202.8, 42.5),
+      clipMode: 'hidden',
+      submorphs: [
+        {
+          type: Label,
+          name: 'padding indicator',
+          borderRadius: 3,
+          fill: Color.rgba(229, 231, 233, 0),
+          fontColor: Color.rgb(101, 135, 139),
+          fontFamily: 'Material Icons',
+          padding: rect(5, 5, 0, 0),
+          textAndAttributes: ['\ue25a', {
+            fontSize: 16,
+            textStyleClasses: ['material-icons']
+          }]
+        },
+        part(NumberWidgetLight, {
+          name: 'padding left',
+          min: 0,
+          extent: pt(40, 22),
+          tooltip: 'Border Radius Top Left',
+          borderRadiusTopRight: 0,
+          borderRadiusBottomRight: 0
+        }),
+        part(NumberWidgetLight, {
+          name: 'padding top',
+          borderWidth: { top: 1, left: 0, right: 0, bottom: 1 },
+          min: 0,
+          borderRadius: 0,
+          extent: pt(40, 22),
+          tooltip: 'Border Radius Top Right'
+        }),
+        part(NumberWidgetLight, {
+          name: 'padding right',
+          borderWidth: { top: 1, left: 1, right: 0, bottom: 1 },
+          min: 0,
+          borderRadius: 0,
+          extent: pt(40, 22),
+          tooltip: 'Border Radius Bottom Right'
+        }),
+        part(NumberWidgetLight, {
+          name: 'padding bottom',
+          min: 0,
+          borderRadiusTopLeft: 0,
+          borderRadiusBottomLeft: 0,
+          extent: pt(40, 22),
+          tooltip: 'Border Radius Bottom Left'
+        })
+      ]
+    }, part(CloseButton, {
+      name: 'independent padding toggle',
+      tooltip: 'Toggle indepentent Fields per Direction',
+      padding: rect(3, 3, 0, 0),
+      position: pt(192.8, 48.6),
+      textAndAttributes: ['', {
+        fontSize: 18,
+        textStyleClasses: ['material-icons']
+      }]
+    })
+  ]
+}
+);
+
+export const PaddingControlsDark = component(PaddingControlsLight, {
```

Comment:
I am not entirely happy with the look of the padding controls yet. I would like this to basically look more like the border radius control in the shape section of the side bar. 

For recap: 
![border radius control](https://user-images.githubusercontent.com/1296388/207594352-77fd179b-ceaf-4698-a9f5-9fe7ece83b81.jpg)

Reply:
Now it looks like this:

![textcontrols](https://user-images.githubusercontent.com/14252419/208492182-6b7aed5d-232c-468c-801d-f1ee337389a3.gif)

Reply:
It's beautiful.